### ISSUE TITLE
Coordinator can not send propose tx when keep `massDeposit` transaction coming in specific situation.

### ISSUE THREAD

**Issue by sifnoc** [zkopru-network/zkopru#223]
I am going to describe a situation that the coordinator could not send propose tx while testing in the local environment.

The testing condition like 1 coordinator and 1 wallet, each has fullnode instance.

I made a script that the wallet node keeps trying to make transfer tx, if it does not have spendable Utxo, the wallet node sends deposit tx then checks spendable and keeps repeating send deposit tx until got available.

In the coordinator, There are enough fees to propose so make tx  then send but failed by a nonce. because coordinator's 'propose worker' just taken a chance to send tx by `commitMassDeposit`.

The coordinator can send `propose` tx successfully in one of 50 blocks in my local environment.

Of course, It is a controlled condition. 
but what if the coordinator could not send well `purpose` or `finalize` transaction by something like `commitMassDeposit`?

**Comment by JChanceHud** [comment]
So there's a race condition between the mass deposit commit worker and the proposal worker? Seems like we should refactor these to operate one after the other. Good catch, this might be what causes the integration tests to fail every now and then.

**Comment by sifnoc** [comment]
> So there's a race condition between the mass deposit commit worker and the proposal worker?
That's right. We should consider the situation that every layer1 block contained a deposit transaction.

Might we need a priority of transactions which can make by coordinator workers?
